yt_dlp/extractor/common.py

```diff
@@ -3489,7 +3489,7 @@ def _parse_jwplayer_formats(self, jwplayer_sources_data, video_id=None,
                 continue
             urls.add(source_url)
             source_type = source.get('type') or ''
-            ext = mimetype2ext(source_type) or determine_ext(source_url)
+            ext = determine_ext(source_url) or mimetype2ext(source_type)
```

Comment:
`determine_ext` defaults to `unknown_video` unless we do this
```suggestion
            ext = determine_ext(source_url, default_ext=None) or mimetype2ext(source_type)
```